File: backend/src/utils/apiResponse.ts
import { Response } from 'express';

export function success(res: Response, data: unknown, statusCode = 200): Response {
  return res.status(statusCode).json({ success: true, data });
}

export function paginated(
  res: Response,
  data: unknown[],
  meta: { total: number; page: number; limit: number }
): Response {
  return res.json({ success: true, data, meta });
}

export function error(res: Response, message: string, statusCode = 400): Response {
  return res.status(statusCode).json({ success: false, message });
}

File: mobile/lib/models/api_response.dart
class ApiResponse<T> {
  final bool success;
  final T? data;
  final String? message;
  final Map<String, dynamic>? meta;

  ApiResponse({
    required this.success,
    this.data,
    this.message,
    this.meta,
  });

  factory ApiResponse.fromJson(
    Map<String, dynamic> json,
    T Function(dynamic)? fromData,
  ) {
    return ApiResponse(
      success: json['success'] as bool,
      data: json['data'] != null && fromData != null
          ? fromData(json['data'])
          : null,
      message: json['message'] as String?,
      meta: json['meta'] as Map<String, dynamic>?,
    );
  }
}
